### DISCUSSION TITLE
Intercepting Firebase Cloud Messaging (FCM) requests/responses?

### DISCUSSION THREAD

**Discussion by dovedevic** [mitmproxy/mitmproxy#6685] [Q&A]
An iOS app I am studying is making form-encoded requests to `https://iid.googleapis.com/iid/register` with specific `X-gcm.topic` and `X-scope` keys. The response is typically just a gzip with decoded raw data `token=`. Using wireshark and rvictl on my Mac, I can see additional application data is being transferred over port 5228, which fits in line with Googles [FCM protocol](https://developers.google.com/cloud-messaging/http). Is there a way to intercept this traffic with mitmproxy, either stock or using a script? I'd like to see what data this app is fetching with these requests.

**Comment by Prinzhorn** [comment]
[`rawtcp`](https://docs.mitmproxy.org/stable/concepts-options/#rawtcp) is enabled by default, so if the app is properly configured to route everything through the proxy (not just HTTP), then you should see non-HTTP stuff.
